src/com/palmergames/bukkit/towny/utils/AreaSelectionUtil.java

```diff
@@ -21,48 +22,61 @@
 
+	/**
+	 * Method to select a List&lt;WorldCoord&gt; of coordinates. 
+	 * Area claims can be either circular or rectangular.
+	 * 
+	 * @param owner - {@link com.palmergames.bukkit.towny.object.TownBlockOwner} making the selection. 
+	 * @param pos - WorldCoord where the selection is being made from.
+	 * @param args - Subcommands like rect, circle, auto or #.
+	 * @return List&lt;WorldCoord&gt; of {@link com.palmergames.bukkit.towny.object.WorldCoord}.
+	 * @throws TownyException - Thrown when invalid subcommands are given.
+	 */
 	public static List<WorldCoord> selectWorldCoordArea(TownBlockOwner owner, WorldCoord pos, String[] args) throws TownyException {
 
 		List<WorldCoord> out = new ArrayList<>();
 
 		if (args.length == 0) {
-			// claim with no sub command entered so attempt selection of one plot
-			if (pos.getTownyWorld().isClaimable())
-				out.add(pos);
-			else
-				throw new TownyException(Translation.of("msg_not_claimable"));
+			/*
+			 * Either /{command} {claim|unclaim} or /town claim {outpost}
+			 */
+			out.add(pos);
+
 		} else {
-			if (args.length > 1) {
+			
+			/*
+			 * First, determine what is available to be claimed.
+			 */
+			int available = 0;
+			if (owner instanceof Town) {
+				Town town = (Town) owner;
+				available = TownySettings.getMaxTownBlocks(town) - town.getTownBlocks().size();
+			} else if (owner instanceof Resident) {
+				available = TownySettings.getMaxResidentPlots((Resident) owner);
+			}
+			
+			/*
+			 * Second, handle different subcommands for /{command} {claim|unclaim} {rect|circle|auto|#} {#}
+			 */
+			if (args.length > 1) { // Has to be /{command} {claim|unclaim} {rect|circle} {auto|#}
 				if (args[0].equalsIgnoreCase("rect")) {
-					out = selectWorldCoordAreaRect(owner, pos, StringMgmt.remFirstArg(args));
+					out = selectWorldCoordAreaRect(available, pos, StringMgmt.remFirstArg(args));
 				} else if (args[0].equalsIgnoreCase("circle")) {
-					out = selectWorldCoordAreaCircle(owner, pos, StringMgmt.remFirstArg(args));
+					out = selectWorldCoordAreaCircle(available, pos, StringMgmt.remFirstArg(args));
 				} else {
 					throw new TownyException(Translation.of("msg_err_invalid_property", StringMgmt.join(args, " ")));
 				}
-			} else if (args[0].equalsIgnoreCase("auto")) {
-				out = selectWorldCoordAreaRect(owner, pos, args);
-			} else if (args[0].equalsIgnoreCase("outpost")) {
-				TownBlock tb = pos.getTownBlock();
-				if (!tb.isOutpost() && tb.hasTown()) { // isOutpost(), only for mysql however, if we include this we can skip the outposts on flatfile so less laggy!
-					Town town = tb.getTown();
-					if (isTownBlockLocContainedInTownOutposts(town.getAllOutpostSpawns(), tb)) {
-						tb.setOutpost(true);
-						out.add(pos);
-					} else {
-						throw new TownyException(Translation.of("msg_err_unclaim_not_outpost"));
-						// Lang String required.
-					}
-				}
-				if (tb.isOutpost()) { // flatfile skipper
-					out.add(pos);
-				}
```

Comment:
Removed because we're doing the same thing anyways when someone does `/t unclaim`